src/store/sagas/tasksSaga.js

```diff
@@ -0,0 +1,48 @@
+import { takeEvery, put } from 'redux-saga/effects';
+import {
+  GET_ALL_TASKS_ASYNC,
+  UPDATE_TASK_ASYNC,
+  DELETE_TASK_BY_ID_ASYNC,
+  SetAllTodos,
+  UpdateTodo,
+  DeleteTodoById
+} from '../actions';
+import {
+  getTodos,
+  deleteTask,
+  updateTask
+} from '../../services';
+
+export function* getTodosAsync() {
+  const tasks = yield getTodos()
+    .catch(err => console.log(err, ' get tasks error'));
+
+  if (tasks !== undefined) {
+    yield put(SetAllTodos(tasks));
+  }
+}
+export function* watchGetTodosAsync() {
+  yield takeEvery(GET_ALL_TASKS_ASYNC, getTodosAsync);
+}
+
+export function* updateTodosAsync(data) {
+  const tasks = yield updateTask(data.task)
+    .catch(err => console.log(err, ' update tasks error'));
```

Comment:
when you're using generators, there is no need to use catch or some other async stuff